File: mock/src/main/java/com/my/test/BcryptConverter.java
package com.my.test;

import org.databene.commons.ConversionException;
import org.databene.commons.converter.ThreadSafeConverter;
import org.mindrot.jbcrypt.BCrypt;

public class BcryptConverter extends ThreadSafeConverter<String, String> {
    public BcryptConverter() {
        super(String.class, String.class);
    }

    public String convert(String s) throws ConversionException {
        return BCrypt.hashpw(s, BCrypt.gensalt());
    }
}

File: seed/src/main/scala/gg/uhc/website/seed/BcryptConverter.scala
package gg.uhc.website.seed

import org.databene.commons.converter.ThreadSafeConverter
import com.github.t3hnar.bcrypt._

class BcryptConverter extends ThreadSafeConverter[String, String](classOf[String], classOf[String]) {
  def convert(s: String): String = s.bcrypt
}
